api/handlers/certificates.go

```diff
@@ -242,9 +245,25 @@ func (h AppHandler) CertCreate(c echo.Context) error {
 			map[string]string{"result": "fail", "message": "Failed parsing signed key", "details": err.Error()})
 	}
 	signedCert := k.(*ssh.Certificate)
+
 	//assigning the new key id to store the new value into db
 	certRequest.CertKeyID = signedCert.KeyId
 	certRequest.SerialNumber = strconv.FormatUint(signedCert.Serial, 10)
+
+	// generate key fingerprint
+	certRequest.KeyFingerprint = ssh.FingerprintSHA256(signedCert.Key)
+
+	certRequest.CertType = signedCert.Type()
+
+	// generate certificate fingerprint
+	// cleanCert example: ssh-rsa-cert-v01@openssh.com AAAAHHNza...
+	cleanCert := strings.SplitN(signedKey, " ", 2)
+	cleanCert[1] = strings.Trim(cleanCert[1], "\n")
+	// cleanCert[1] AAAAHHNza...=
+	sha256sum := sha256.Sum256([]byte(cleanCert[1]))
+	hash := base64.RawStdEncoding.EncodeToString(sha256sum[:])
+	certRequest.CertFingerprint = hash
```

Comment:
Just to mention, I would rather create a function to generate this hash instead of adding all this logic here.

Reply:
Clean code... Done!